language/rescue_spec.rb

```diff
@@ -23,11 +23,22 @@ class ArbitraryException < StandardError
     end.should == :caught
   end
 
-  it "can capture the raised exception in a local variable" do
-    begin
-      raise SpecificExampleException, "some text"
-    rescue SpecificExampleException => e
-      e.message.should == "some text"
+  {
+    # Standard use case
+    'can capture the raised exception in a local variable'                    => RescueSpecs::LocalVariableCaptor,
+    # Exotic use cases
+    'can capture the raised exception in a class variable'                    => RescueSpecs::ClassVariableCaptor,
+    'can capture the raised exception in a constant'                          => RescueSpecs::ConstantCaptor,
+    'can capture the raised exception in a global variable'                   => RescueSpecs::GlobalVariableCaptor,
+    'can capture the raised exception in an instance variable'                => RescueSpecs::InstanceVariableCaptor,
+    'can capture the raised exception using a safely navigated setter method' => RescueSpecs::SafeNavigationSetterCaptor,
+    'can capture the raised exception using a setter method'                  => RescueSpecs::SetterCaptor,
+    'can capture the raised exception using a square brackets setter'         => RescueSpecs::SquareBracketsCaptor,
```

Comment:
Since most cases would be almost identical, but some required a supporting object on which to set a class/instance variable, or call a setter method, I decided to implement all cases using supporting objects with a shared API, and list them out concisely.

Reply:
In general we try to avoid dynamically-generated specs in ruby/spec.
I think a middle ground here would be to still to have 1 `it` per case, but share the logic with the `should` etc in a method, which you can define here under the describe:
```ruby
  def should_capture(captor)
    captor.capture('some text').should == :caught # Ensure rescue body still runs
    captor.captured_error.message.should == 'some text'
  end

  it 'can capture the raised exception in a local variable' do
    should_capture(RescueSpecs::LocalVariableCaptor.new)
  end

  ...
```

Reply:
The common part `can capture the raised exception` can also be factored out with an extra `describe` around.

Reply:
@sambostock Could you do that change? Then I think it's ready to merge.

Reply:
@eregon I've pushed a commit that does something similar to your suggestion.

```ruby
describe 'can capture the raised exception' do
  it 'in a local variable' do
    RescueSpecs::LocalVariableCaptor.should_capture_exception
  end
```
Let me know what you think!